### ISSUE TITLE
Removal of "swagger-ui.html" from uri path causes swagger-ui JavaScript error

### ISSUE THREAD

**Issue by LukeHackett** [springfox/springfox#1707]
- **What version of the library are you using?** 
v2.6.1

 - [x] **Bug report**
 
Application Stack
- Spring Boot 1.5 (with spring web)
- Swagger 2.6.1

We're looking to integrate with springfox-ui with Spring Boot, but without having "swagger-ui.html" within the context path. For example instead of having a url that looks like `http://localhost:8080/some-application/swagger-ui.html` we ideally would like a url that looks like `http://localhost:8080/some-application/`

We are able to achieve this using a Spring controller, that simply loads the swagger-ui view:
```
@ApiIgnore
@Controller
@RequestMapping("/")
public class SwaggerController {

    @GetMapping
    public String index() {
        return "swagger-ui.html";
    }

}
```
but upon loading the page, we receive a JavaScript error (see attached).
![console](https://cloud.githubusercontent.com/assets/1360064/23654649/9131edf8-0329-11e7-8421-918d1a0ff8f0.png)
![swagger-ui](https://cloud.githubusercontent.com/assets/1360064/23654651/934f526a-0329-11e7-822a-91e453912dae.png)

Basically, as "swagger-ui.html" is not found in the url at all, undefined is returned ([line 4 ; springfox.js](https://github.com/springfox/springfox/blob/master/springfox-swagger-ui/src/web/js/springfox.js#L3-L6)), resulting in an error on  ([line 5 ; springfox.js](https://github.com/springfox/springfox/blob/master/springfox-swagger-ui/src/web/js/springfox.js#L3-L6)).

The code doesn't handle this use case, but can easily be made to handle the use case. I wanted to make sure that this is a bug (and that there isn't an option / trick I'm missing). I'm happy to raise a PR to fix the issue if it is a bug.

**Comment by dilipkrish** [comment]
Removal of swagger-ui.html is not supported. It currently uses the location of that file to determine where the api-docs are available

**Comment by dilipkrish** [comment]
<img width="449" alt="swagger_ui" src="https://cloud.githubusercontent.com/assets/73257/23719596/a7527c28-0401-11e7-861b-b753a850ecc2.png">

Hope this helps!
